extension/app/ts/background/windows/interceptorAccess.ts

```diff
@@ -137,114 +133,138 @@ export async function requestAccessFromUser(port: browser.runtime.Port | undefin
 		const message = PopupMessage.parse(msg)
 		if (message.method !== 'popup_interceptorAccessReadyAndListening') return
 		browser.runtime.onMessage.removeListener(windowReadyAndListening)
-		if (globalThis.interceptor.settings === undefined) return rejectReply
+		if (globalThis.interceptor.settings === undefined) return rejectReply()
 		return await sendPopupMessageToOpenWindows({
 			method: 'popup_interceptorAccessDialog',
 			data: {
 				website: website,
 				requestAccessToAddress: accessAddress,
+				originalRequestAccessToAddress: accessAddress,
 				associatedAddresses: associatedAddresses,
 				addressInfos: globalThis.interceptor.settings.userAddressBook.addressInfos,
 				signerAccounts: [],
 				signerName: globalThis.interceptor.signerName,
-				simulationMode: simulationMode,
-				allowAddressChanging: port !== undefined,
+				simulationMode: globalThis.interceptor.settings.simulationMode,
+				socket: socket,
 			}
 		})
 	}
 
 	if (pendingInterceptorAccess !== undefined) {
-		if (pendingInterceptorAccess.websiteOrigin === website.websiteOrigin && pendingInterceptorAccess.requestAccessToAddress === accessAddress?.address) {
-			return rejectReply // there's already one pending request, and it's different access request
+		return rejectReply()
+	}
+
+	pendingInterceptorAccess = {
+		future: new Future<InterceptorAccessReply>(),
+		websiteOrigin: website.websiteOrigin,
+		requestAccessToAddress: accessAddress?.address,
+	}
+
+	const oldPromise = await getPendingInterceptorAccessRequestPromise()
+	if (oldPromise !== undefined) {
+		if ((await chrome.tabs.query({ windowId: oldPromise.dialogId })).length > 0) {
+			pendingInterceptorAccess = undefined
+			return rejectReply()
 		}
-	} else {
-		pendingInterceptorAccess = {
-			future: new Future<InterceptorAccessOptions>(),
-			websiteOrigin: website.websiteOrigin,
-			requestAccessToAddress: accessAddress?.address,
+		await savePendingInterceptorAccessRequestPromise(undefined)
+	}
+
+	browser.runtime.onMessage.addListener(windowReadyAndListening)
+
+	openedInterceptorAccessWindow = await browser.windows.create(
+		{
+			url: getHtmlFile('interceptorAccess'),
+			type: 'popup',
+			height: 600,
+			width: 600,
```

Comment:
This seems unnecessarily small, and I don't think will ever fit everything.  Can we make the default bigger?